impeller/aiks/experimental_canvas.cc

```diff
@@ -431,6 +432,14 @@ void ExperimentalCanvas::SaveLayer(
   entry.rendering_mode = Entity::RenderingMode::kSubpassAppendSnapshotTransform;
   transform_stack_.emplace_back(entry);
 
+  // The DisplayList bounds/rtree doesn't account for filters applied to parent
+  // layers, and so sub-DisplayLists are getting culled as if no filters are
+  // applied.
+  // See also: https://github.com/flutter/flutter/issues/139294
+  if (paint.image_filter) {
```

Comment:
Once we switch to exp canvas we can delete the cull rect and just use the clip stack which is 99% the same data.

Reply:
I think the intention here is that if something outside of a DL applies an image filter then you need to cull the sub-dl by the "GetSourceCoverage" of that filter. No, it doesn't take that into account, the caller specifying the cull rect was supposed to...

Reply:
Oh, wait, layers. We do expand the bounds of sub-layers by a filter applied on a parent layer...

https://github.com/flutter/engine/blob/09fa10157ca839e27e732f943f0b63b4fdc26be0/display_list/dl_builder.cc#L680

If rtree, then here:
https://github.com/flutter/engine/blob/09fa10157ca839e27e732f943f0b63b4fdc26be0/display_list/dl_builder.cc#L722

And if no rtree, then here:
https://github.com/flutter/engine/blob/09fa10157ca839e27e732f943f0b63b4fdc26be0/display_list/dl_builder.cc#L730

Reply:
This comment isn't really accurate. I updated it to document that the cull rect logic gets deleted with new canvas.